### ISSUE TITLE
Should throw error when export an external declaration in a module

### ISSUE THREAD

**Issue by cqzooge** [microsoft/TypeScript#51708]
# Bug Report

### 🔎 Search Terms

module, export

### 🕗 Version & Regression Information

- This is the behavior in every version I tried, and I reviewed the FAQ for entries about export, but I don't find the answer

### ⏯ Playground Link

https://www.typescriptlang.org/play

### 💻 Code

```ts
var a = 1;
declare namespace ns {
    export {a};
}

var b=1;
export default b;
```
```ts
var a = 1;
declare namespace ns {
    export {a};
}
```

### 🙁 Actual behavior

The first codes don't throw error.
And the second codes throw the error: `Cannot export 'a'. Only local declarations can be exported from a module.`

### 🙂 Expected behavior

I think both of them should have same behavier: throw error: `Cannot export 'a'. Only local declarations can be exported from a module.`.
I don't know how the `export default` affect the `export` in namespace. Is this a bug or not? I don't find describe in FAQs or spec.
And I found I can use `export = ...` replace `export default`. Are they same problems?

**Comment by RyanCavanaugh** [comment]
The error means "local" as "not global". In the first case, it's not a global, so the error doesn't happen.
